socorro/__init__.py

```diff
@@ -1,3 +1,116 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import contextlib
+import copy
+import importlib
+import logging
+import os
+import re
+
+import glom
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+UPPERCASE_KEY_RE = re.compile(r"^[A-Z][A-Z_]+$")
+
+
+class Settings:
+    def __init__(self, settings_module_path):
+        self._settings_module_path = settings_module_path
+
+        settings_module = importlib.import_module(settings_module_path)
+        for key in dir(settings_module):
+            if not UPPERCASE_KEY_RE.match(key):
+                continue
+            setattr(self, key, getattr(settings_module, key))
+
+    def log_settings(self, logger=LOGGER):
+        """Log settings making sure to sanitize values that have "secret" in the key"""
+
+        def sanitize_value(k, v):
+            if "secret" in k.lower():
+                return "***** (secret)"
+
+            if isinstance(v, (tuple, list)):
+                return type(v)([sanitize_value(k, v_item) for v_item in v])
+
+            if isinstance(v, dict):
+                return {
+                    sub_k: sanitize_value(sub_k, sub_v) for sub_k, sub_v in v.items()
+                }
+            return v
+
+        for key in dir(self):
+            if not UPPERCASE_KEY_RE.match(key):
+                continue
+            value = sanitize_value(key, getattr(self, key))
+
+            logger.info("%s: %r", key, value)
+
+    def __repr__(self):
+        return f"<Settings {self._settings_module_path!r}>"
+
+    @contextlib.contextmanager
+    def override(self, **kwargs):
```

Comment:
This needs usage examples.

Reply:
This could probably use some tests, too. Does it build up correctly? Does it override correctly? Does it take things down correctly?